client/txn.go

```diff
@@ -78,18 +78,37 @@ type Txn struct {
 	haveEndTxn   bool // True if there was an explicit EndTransaction
 }
 
-func newTxn(db DB, depth int) *Txn {
+func newTxn(db DB) *Txn {
 	txn := &Txn{
 		db:      db,
 		wrapped: db.Sender,
 	}
 	txn.db.Sender = (*txnSender)(txn)
 
-	file, line, fun := caller.Lookup(depth + 1)
+	// Caller's caller.
+	file, line, fun := caller.Lookup(2)
 	txn.txn.Name = fmt.Sprintf("%s:%d %s", file, line, fun)
 	return txn
 }
 
+// NewTxn returns a new txn.
+func NewTxn(db DB) *Txn {
```

Comment:
this isn't what i meant - this will result in all sql transactions having the same name